mypy/newsemanal/semanal.py

```diff
@@ -2721,8 +2726,15 @@ def process_typevar_declaration(self, s: AssignmentStmt) -> bool:
         # Yes, it's a valid type variable definition! Add it to the symbol table.
         if existing and isinstance(existing.node, TypeVarExpr):
             # Existing definition from previous semanal iteration, use it.
-            # TODO: This may be confused with a duplicate TypeVar definition.
-            #       Fix this and add corresponding tests.
+            # Note that unlike in most other cases, we patch an existing symbol, instead of
+            # adding the type variable symbol only when the types it depends on are ready.
+            # This is necessary to avoid a "deadlock" in this typical situation:
+            #     T = TypeVar('T', bound=C[Any])
+            #     class C(Generic[T]):
+            #         ...
+            # We can use allow_placeholder=True for upper bound and values to avoid this,
```

Comment:
This doesn't make it clear whether we are actually doing this. Instead, write something like "We could use ..., but we don't do this, because ...".

What happens if the type variable definition and the class definition are in different parts of an import cycle? Do we ensure that once the type variable has been completed, the new bound will be used in the class definition?